app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,33 @@ def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def new
+    puts('life in new')
+    @user = ApplicationController.new.current_user
+    @post = @user.posts.new
+    render :new, locals: { post: @post }
+  end
+
+  def create
+    @user = ApplicationController.new.current_user
+    add_post = @user.posts.new(post_params)
+    respond_to do |format|
```

Comment:
- [ ] in these two methods you are repeating code, this can be written differently.
- in the new method you only have to set the @post instance variable `@post = Post.new`
- in the create method you can use current_user as follows: `@post = @current_user.posts.new(post_params)`